### DISCUSSION TITLE
merging PDFs loses attachment

### DISCUSSION THREAD

**Discussion by KRISHNAVIB** [LibrePDF/OpenPDF#1144] [General]
Hi There,
I was trying to merge two PDFs in which one of the PDF is having attachment (docx file). After merging both PDFs using OpenPDF , we can see the attachment which was there in one of the PDF is not in the final output file i.e: Merged File.
IS OpenPDF capable of handling attachments while merging PDFs.
Can anyone help?

**Comment by mkl-public** [comment]
Merging PDFs with `PdfCopy` only merges page-level material from the source PDFs, not document-level material like general file attachments. If you want to merge PDFs and keep document-level material from one of the documents, open that file in a `PdfStamper` and use `importPage` to add the pages from the other documents.

**Reply by KRISHNAVIB** [reply]
Tried that, but that doesn't carry attachment. Can you elaborate?
Thanks
